t/apicast-policy-camel.t

```diff
@@ -315,3 +315,193 @@ ETag: foobar
 <<EOF
 using proxy: http://127.0.0.1:$Test::Nginx::Util::PROXY_SSL_PORT,
 EOF
+
+
+=== TEST 5: API backend connection uses http proxy with Basic Auth
```

Comment:
I wonder how this works with the camel proxy when upstream is TLS. 

You are adding the `Proxy-Authorization` header to the method `_connect_proxy_https`. But precisely, for the camel proxies, that method is not being used. Instead, `_connect_tls_direct` is being used. Correct me if I am wrong.

Does it even make sense to add authentication support on camel proxy when upstream is TLS?

Reply:
Now I see your comment

```
The camel proxy policy is interesting here as we sent over HTTPS but skip CONNECT, so I wonder what is the correct approach here

    Set Proxy-Authorization header in the original request
    Don't set the header and update the documentation saying that camel proxy policy is incompatible with proxy authentication
```

Regarding `Set Proxy-Authorization header in the original request`, IMO it does not make sense to me. The header would not even be read by camel proxy and would end up in the upstream. 

Regarding `Don't set the header and update the documentation saying that camel proxy policy is incompatible with proxy authentication`, we may need to investigate further how camel proxy works and if they define a use case for proxy auth on TLS connections. Maybe here? https://github.com/zregvart/camel-netty-proxy They say it is only an example.

Reply:
In anyway, the integration tests for the TLS upstream does not make sense as it is for now.